Add support for single-station and historical MOSMIX files

FabianVolkers opened the pull request:
- changed the MOSMIXParser regex to also match single station mosmix files

FabianVolkers committed:
✨ Update MOSMIXParser regex to support single station files

jdemaeyer commented:
Hi @FabianVolkers, thanks for the PR!

This tight MOSMIX regex is a leftover from when this code was still part of `brightsky`. You're right, the restriction to the "all stations" file doesn't make sense within `dwdparse` anymore. Neither does the restriction to the `_LATEST` file. We should probably change the regex to just `r'MOSMIX_'`?

FabianVolkers committed:
✨ Update MOSMIX regex to remove restriction to `LATEST` file

FabianVolkers committed:
✅ Update `test_get_parser` to test all MOSMIX filenames

FabianVolkers commented:
Hi @jdemaeyer, thanks for your reply.

It would make sense to remove the restriction to the `_LATEST` file. I hadn't tested that. I've updated the PR to change the regex to just `r'MOSMIX_'`. I also added tests for the different types of MOSMIX filenames in `test_get_parser`.

Do we also want update `test_mosmix_parser` to test the MOSMIX parser with the different files?

jdemaeyer commented:
Perfect, thanks! Let's leave it at the `get_parser` test for now. The MOSMIX test fixtures are a little annoying to set up since they require modifying the (compressed) data files so we don't bloat the repo, and since we don't have a specific single-location edge case to test I think we can get away with leaving the existing tests untouched.

I'll merge and create a release in a bit!
